src/DynamoCore/Engine/CodeGeneration/IAstNodeContainer.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using ProtoCore.AST.AssociativeAST;
+
+namespace Dynamo.Engine.CodeGeneration
+{
+    /// <summary>
+    /// Returns notification for AST compilation events.                                                 
+    /// </summary>
+    public interface IAstNodeContainer
+    {
+        /// <summary>
+        /// Indicates to start compiling a NodeModel to AST nodes.
+        /// </summary>
+        /// <param name="nodeGuid"></param>
+        void OnCompiling(Guid nodeGuid);
+
+        /// <summary>
+        /// Indicates a NodeModel has been compiled to AST nodes. 
+        /// </summary>
+        /// <param name="nodeGuid"></param>
+        /// <param name="astNodes"></param>
+        void OnCompiled(Guid nodeGuid, IEnumerable<AssociativeNode> astNodes);
```

Comment:
can we use `Node` instead of `AssociativeNode` ?

Reply:
I don't think so, due to this being a previously public interface. Not sure if that would break binary compatibility. An overload could be added and this obsoleted, but it could be potentially confusing.